### PR TITLE
Merge develop & fix YOLOX inference in BFP16 on XPU

### PR DESCRIPTION
**Pull request by sovrasov**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py [hunk 1]
```diff
@@ -21,6 +21,88 @@ class CustomYOLOXHead(YOLOXHead):
...
+        pos_masks = torch.cat(pos_masks, 0)
+        cls_targets = torch.cat(cls_targets, 0)
+        obj_targets = torch.cat(obj_targets, 0)
+        bbox_targets = torch.cat(bbox_targets, 0)
+        if self.use_l1:
+            l1_targets = torch.cat(l1_targets, 0)
+
+        loss_bbox = self.loss_bbox(flatten_bboxes.view(-1, 4)[pos_masks], bbox_targets) / num_total_samples
+        loss_obj = self.loss_obj(flatten_objectness.view(-1, 1), obj_targets) / num_total_samples
+        if "xpu" in str(flatten_cls_preds.device):
```
Comment (eunwoosh) [inline src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py:88]: Does `view` make an error in XPU?

**Review by eunwoosh — COMMENTED** [review]
Thanks for your work. I have one question. Please take a look.

File: src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py (hunk 1 above)
Comment (sovrasov) [reply src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py:88]: Yes, XPU doesn't support view for that particular case, so I had to use reshape here

**Comment by kprokofi** [comment]
Shouldn't we update (rebase / merge) regularly the `develop-idev` branch to the latest develop instead of incoming branches? What do you think?

**Comment by sungchul1** [comment]
> Shouldn't we update (rebase / merge) regularly the `develop-idev` branch to the latest develop instead of incoming branches?

I agree. Updating `develop-idev` would be better to check differences on this branch.

File: src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py (hunk 1 above)
Comment (kprokofi) [reply src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py:88]: maybe we just substitute view to reshape for all devices? Are there any issues or concerns regarding this operation?

Comment (sovrasov) [reply src/otx/algorithms/detection/adapters/mmdet/models/heads/custom_yolox_head.py:88]: View doesn't allocate a new tensor, it just creates a mapping to the existing memory (which is preferable), while reshape requires more memory.

**Review by kprokofi — APPROVED** [review]

**Review by eunwoosh — APPROVED** [review]

**Review by sungchul1 — APPROVED** [review]

**Pull request merged**